metastore/src/java/org/apache/hadoop/hive/metastore/SerDeStorageSchemaReader.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.serde2.Deserializer;
+import org.apache.hadoop.hive.metastore.api.EnvironmentContext;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.utils.StringUtils;
+
+import java.util.List;
+
+public class SerDeStorageSchemaReader implements StorageSchemaReader {
```

Comment:
where is this used ?

Reply:
HiveMetaStore.get_fields_with_environment_context().  Previously this method used the SerDe to read the fields from the SerDe parameters rather than the storage descriptor fields.  However, standalone-metastore doesn't have access to the serdes.  We haven't yet settled on a solution for this.  There's a debate raging on HIVE-17714 on what the right way forward is.  For now I've created the StorageSchemaReader interface to pull the serde dependency out.  There is a default implementation that can be used by the metastore in standalone mode that just fails if its called.  SerDeStorageSchemaReader is the implementation for use with Hive that works as before.